File: sdks/typescript/src/types/workspaces.ts
/**
 * Workspace and tenant types.
 *
 * @module types/workspaces
 * @see edgequake/crates/edgequake-api/src/handlers/workspaces_types.rs
 */

import type { Timestamp } from "./common.js";

// ── Tenants ───────────────────────────────────────────────────

export interface CreateTenantRequest {
  name: string;
  metadata?: Record<string, unknown>;
}

export interface TenantInfo {
  tenant_id: string;
  name: string;
  created_at: Timestamp;
}

export interface TenantDetail extends TenantInfo {
  workspace_count: number;
  metadata?: Record<string, unknown>;
}

export interface TenantResponse extends TenantInfo {}

export interface UpdateTenantRequest {
  name?: string;
  metadata?: Record<string, unknown>;
}

// ── Workspaces ────────────────────────────────────────────────

export interface CreateWorkspaceRequest {
  name: string;
  slug?: string;
  description?: string;
  metadata?: Record<string, unknown>;
}

export interface WorkspaceInfo {
  workspace_id: string;
  tenant_id: string;
  name: string;
  slug: string;
  created_at: Timestamp;
}

export interface WorkspaceDetail extends WorkspaceInfo {
  description?: string;
  metadata?: Record<string, unknown>;
  document_count?: number;
  entity_count?: number;
}

export interface WorkspaceResponse extends WorkspaceInfo {}

export interface UpdateWorkspaceRequest {
  name?: string;
  slug?: string;
  description?: string;
  metadata?: Record<string, unknown>;
}

export interface WorkspaceStats {
  workspace_id: string;
  document_count: number;
  entity_count: number;
  relationship_count: number;
  chunk_count: number;
  storage_bytes?: number;
}

export interface MetricsHistoryQuery {
  from?: string;
  to?: string;
  interval?: string;
}

export interface MetricsHistory {
  workspace_id: string;
  data_points: Array<{
    timestamp: Timestamp;
    document_count: number;
    entity_count: number;
    relationship_count: number;
  }>;
}

File: sdks/rust/src/types/workspaces.rs
//! Workspace types.

use serde::{Deserialize, Serialize};

/// Create workspace request.
#[derive(Debug, Clone, Serialize)]
pub struct CreateWorkspaceRequest {
    pub name: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub slug: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,
}

/// Workspace summary.
#[derive(Debug, Clone, Deserialize)]
pub struct WorkspaceInfo {
    pub id: String,
    pub name: String,
    #[serde(default)]
    pub slug: Option<String>,
    #[serde(default)]
    pub description: Option<String>,
    #[serde(default)]
    pub tenant_id: Option<String>,
    #[serde(default)]
    pub created_at: Option<String>,
}

/// Workspace statistics.
#[derive(Debug, Clone, Deserialize)]
pub struct WorkspaceStats {
    pub workspace_id: String,
    #[serde(default)]
    pub document_count: u32,
    #[serde(default)]
    pub entity_count: u32,
    #[serde(default)]
    pub relationship_count: u32,
    #[serde(default)]
    pub chunk_count: u32,
    #[serde(default)]
    pub query_count: u32,
    #[serde(default)]
    pub storage_size_bytes: u64,
}

/// Rebuild response.
#[derive(Debug, Clone, Deserialize)]
pub struct RebuildResponse {
    pub status: String,
    #[serde(default)]
    pub message: Option<String>,
    #[serde(default)]
    pub track_id: Option<String>,
}
